docs/guides/src/main/server/run-containers.adoc

```diff
@@ -0,0 +1,106 @@
+ENV KEYCLOAK_ADMIN=admin
+ENV KEYCLOAK_ADMIN_PASSWORD=change_me
+# change these values to point to a running postgres instance
+ENV KC_DB_URL=<DBURL>
+ENV KC_DB_USERNAME=<DBUSERNAME>
+ENV KC_DB_PASSWORD=<DBPASSWORD>
+ENV KC_HOSTNAME=localhost:8443
+ENTRYPOINT ["/opt/keycloak/bin/kc.sh", "start"]
+----
+In the first stage of this multi-staged build, we're creating the optimized image using the `build` command to apply the build options. From the `builder`, we're copying the files generated by the `build` process into a new image. In this runner image, we apply the specific run configuration, containing e.g. a keystore, the environment-specific hostname configuration and database configuration. Then this image gets started in production mode by using the `start` command in the entrypoint.
+
+Note that this example uses a multi-staged build to explicitly showcase the build and run steps. It could also be run as a single-staged docker build.
+
+=== Building the docker image
+To build the actual docker image, run the following command form the Directory containing your Dockerfile:
+[source,bash]
+----
+podman|docker build . -t prebuilt_keycloak
+----
+
+=== Start the optimized Keycloak docker image:
+To start the image, run:
+[source, bash]
+----
+podman|docker run --rm -it --name optimized_keycloak -p 8443:8443 prebuilt_keycloak
+----
+
+You'll notice that the startup log contains the following line:
+[source, bash]
+----
+INFO  [org.key.com.Profile] (main) Preview feature enabled: token_exchange
+----
+stating that the desired feature is enabled.
+Also, opening up `https://localhost:8443/metrics` leads to a page containing operational metrics which could be used by your monitoring solution.
+
+== Try Keycloak out in development mode
+The easiest way to try out Keycloak from a container for development or testing purposes is using the Development mode(_todo_link_to_missing_guide_) by using the `start-dev` command:
+
+[source,bash]
+----
+podman|docker run --rm -it --name keycloak_test -p 8080:8080 \
+        -e KEYCLOAK_ADMIN=admin -e KEYCLOAK_ADMIN_PASSWORD=change_me \
+        quay.io/keycloak/keycloak-x:latest \
+        start-dev
+----
+
+Invoking this command will spin up the Keycloak server in development mode.
+
+Keep in mind that this mode is by no means meant to be used in production environments, as it has insecure defaults. For more information about running Keycloak in production, see _todo_link_to_running_in_production_guide_.
+
+== Use auto-build to run a standard keycloak container
+Following concepts such as immutable infrastructure, containers need to be re-provisioned fairly often. In these environments, you want to have containers which start up fast by using an optimized image as described above.
+
+However, when your environment is different, you may want to run a standard Keycloak image using the `--auto-build` flag like below:
+
+[source, bash]
+----
+podman|docker run --rm -it --name keycloak_auto_build -p 8080:8080 \
```

Comment:
I'd say we should rather favour environment variables as that is more the standard container way?

Reply:
Wanted to give one example to show that you _can_ use the cli commands too if you wanted to. But not hard on that, I could also change this to environment variables. wdyt?

Reply:
decided to leave as is for now.